src/test/java/uk/gov/hmcts/ccd/domain/model/common/DisplayContextParameterUtilTest.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.hmcts.ccd.domain.model.common;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static java.util.Arrays.asList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static uk.gov.hmcts.ccd.domain.model.common.DisplayContextParameterUtil.Parameter.COLLECTION;
+import static uk.gov.hmcts.ccd.domain.model.common.DisplayContextParameterUtil.Parameter.TABLE;
+import static uk.gov.hmcts.ccd.domain.model.common.DisplayContextParameterUtil.updateDisplayContextParameter;
+
+class DisplayContextParameterUtilTest {
+
+    @Test
+    @DisplayName("should create parameter)")
+    void shouldCreateParameter() {
+        assertAll(
+            () -> assertThat(updateDisplayContextParameter("", COLLECTION, asList("allowInsert", "allowDelete")),
+                is("#COLLECTION(allowDelete,allowInsert)")),
+            () -> assertThat(updateDisplayContextParameter("", COLLECTION, asList("allowDelete")),
+                is("#COLLECTION(allowDelete)")),
+            () -> assertThat(updateDisplayContextParameter("", TABLE, asList("postcode")),
+                is("#TABLE(postcode)")),
+            () -> assertThat(updateDisplayContextParameter("", TABLE, asList("AddressLine1", "postcode")),
+                is("#TABLE(AddressLine1,postcode)"))
+                 );
+    }
+
+    @Test
+    @DisplayName("should update parameter)")
+    void shouldUpdateParameter() {
+        assertAll(
+            () -> assertThat(updateDisplayContextParameter("#COLLECTION(allowInsert,allowDelete)", COLLECTION, asList("allowInsert", "allowDelete")),
+                is("#COLLECTION(allowDelete,allowInsert)")),
+            () -> assertThat(updateDisplayContextParameter("#COLLECTION(allowInsert,allowDelete)", COLLECTION, asList("allowDelete")),
+                is("#COLLECTION(allowDelete)")),
+            () -> assertThat(updateDisplayContextParameter("#COLLECTION()", COLLECTION, asList("allowInsert")),
+                is("#COLLECTION(allowInsert)")),
```

Comment:
can you add a scenario:
`"#COLLECTION("someOtherOption")", COLLECTION, asList("allowInsert")`
and verify that the end result is 
`"#COLLECTION("someOtherOption","allowInsert" )"`
we don't want "someOtherOption" to be wiped out, because it could be some other display option for collections we introduce in the future

Reply:
can you add a scenario for:
"#COLLECTION(allowInsert,allowDelete)", COLLECTION, asList("allowDelete", "allowInsert")

Reply:
ok, changing the behaviour to preserve other properties

Reply:
both done 👍